### PR TITLE
fix: dubbo's thread pool is used when the triple protocol is invoked (#11867)

### PR DESCRIPTION
**Pull request by robin977**
fix: dubbo's thread pool is used when the triple protocol is invoked (#11867)

### CODE REVIEW

File: dubbo-metrics/dubbo-metrics-prometheus/src/test/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricesThreadPoolTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
+package org.apache.dubbo.metrics.prometheus;
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-prometheus/src/test/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricesThreadPoolTest.java:1]: please add apache licence

File: dubbo-metrics/dubbo-metrics-prometheus/src/test/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricesThreadPoolTest.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import com.sun.net.httpserver.HttpServer;
+import org.apache.dubbo.config.*;
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-prometheus/src/test/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricesThreadPoolTest.java:3]: please remove  *

File: dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/ThreadPoolRejectMetric.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ */
+
+package org.apache.dubbo.metrics.model;
+
+import org.apache.dubbo.common.utils.ConfigUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import static org.apache.dubbo.common.constants.MetricsConstants.*;
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/ThreadPoolRejectMetric.java:25]: please  remove  *

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadRejectMetricsCountSampler.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package org.apache.dubbo.metrics.collector.sample;
+
+import org.apache.dubbo.common.utils.ConcurrentHashSet;
+import org.apache.dubbo.metrics.collector.DefaultMetricsCollector;
+import org.apache.dubbo.metrics.model.*;
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadRejectMetricsCountSampler.java:5]: please remove  *

**Review by songxiaosheng — CHANGES_REQUESTED** [review]
please fix some  ci  problem

**Review by AlbumenJ — CHANGES_REQUESTED** [review]
What is the purpose of this PR? For bugfix or add metrics?

**Comment by songxiaosheng** [comment]
please add apache licence header
image

**Pull request closed without merging**